src/crypto.rs

```diff
@@ -60,13 +57,58 @@ impl Pseudonymize for DefaultHasher {
             Literal::LanguageTaggedString { value, language: _ } => value,
             Literal::Simple { value } => value,
         };
-        let hash = blake3::hash(value.as_bytes());
-        return Literal::Simple {
-            value: hash.to_string(),
-        };
+        let crypted = self.pseudo(value.as_bytes());
+        return Literal::Simple { value: crypted };
     }
 
     fn pseudo_blank_node(&self, u: &BlankNode) -> BlankNode {
         return u.clone();
     }
 }
+
+// Helper enums to instantiate various hashers
+
+pub enum Algorithm {
+    Blake3,
+}
+
+impl Default for Algorithm {
+    fn default() -> Self {
+        return Algorithm::Blake3;
+    }
+}
+
+pub fn get_pseudonymizer(algo: Option<Algorithm>, key: Option<Vec<u8>>) -> impl Pseudonymize {
+    let secret = key.unwrap_or(generate_key(32));
+
+    let pseudonymizer = match algo.unwrap_or_default() {
+        Algorithm::Blake3 => Blake3Hasher::new(secret),
+    };
+
+    return pseudonymizer;
+}
+
+// BLAKE3 hasher
+
+pub struct Blake3Hasher {
+    pub key: [u8; 32],
+}
+
+impl Blake3Hasher {
+    pub fn new(key: Vec<u8>) -> Self {
+        let mut new_key = [0u8; 32];
+        if key.len() == 32 {
+            new_key.copy_from_slice(&key[..32]);
+        } else {
+            panic!("Key must be 32 bytes long");
+        }
+
+        return Self { key: new_key };
+    }
+}
```

Comment:
- Change input parameter to `Optional<Vec<u8>>`
- Move `key.unwrap_or(...)` here
- Check minimal key length depending on the algorithm
- Hash key to have the correct key length

Reply:
Good points: all are now implemented